eval.c

```diff
@@ -804,44 +806,37 @@ static VALUE
 make_exception(int argc, const VALUE *argv, int isstr)
 {
     VALUE mesg, exc;
-    int n;
 
     mesg = Qnil;
     switch (argc) {
       case 0:
-	break;
+        return Qnil;
       case 1:
 	exc = argv[0];
-	if (NIL_P(exc))
-	    break;
-	if (isstr) {
+        if (isstr &&! NIL_P(exc)) {
 	    mesg = rb_check_string_type(exc);
 	    if (!NIL_P(mesg)) {
 		mesg = rb_exc_new3(rb_eRuntimeError, mesg);
-		break;
```

Comment:
As `rb_exc_new3` should return an exception instance, isn't ok to return it here?

Reply:
:+1: Yes, that's a nice idea.

Reply:
https://github.com/ruby/ruby/pull/3247/commits/c0bc23f6e0fe0628da8fd89d8ba31fbd89d339ff